doc/source/examples/extended_examples/Krylov/krylov_example.rst

```diff
@@ -0,0 +1,378 @@
+
+
+
+Harmonic analysis using the Frequency sweep Krylov method
+=========================================================
+
+This example shows how to use the krylov method in `PyMAPDL <https://mapdl.docs.pyansys.com/>`_ for harmonic
+analysis.
+
+These are the main steps required:
+
+-  Generate a Krylov subspace for model reduction in harmonic analysis.
+   Using :func:`KrylovSolver.krygensub() <ansys.mapdl.core.krylov.KrylovSolver.krygensub>` method.
+
+-  Reduce the system of equations and solve at each frequency.
+   Using :func:`KrylovSolver.krysolve() <ansys.mapdl.core.krylov.KrylovSolver.krysolve>` method.
+
+-  expand reduced solution back to FE space
+   Using :func:`KrylovSolver.kryexpand() <ansys.mapdl.core.krylov.KrylovSolver.kryexpand>` method.
+
+Problem Description
+-------------------
+
+To perform Harmonic analysis on a cylindrical acoustic duct using the
+`Krylov method <https://en.wikipedia.org/wiki/Krylov_subspace>`_ and study the response of the system over a range of
+frequencies.
+
+The model is a cylindrical acoustic duct with pressure load on one end
+and output impedance on the other end.
+
+.. code:: ipython3
+
+    import os
+    import numpy as np
+    import math
+    import matplotlib.pyplot as plt
+    from ansys.mapdl.core import launch_mapdl
+
+Launch mapdl in ``'-smp'`` mode to avoid distributing processing which is not supported by the MAPDL Math module.
+
+.. code:: ipython3
+
+    mapdl = launch_mapdl(additional_switches='-smp', nproc=4)
+    mapdl.clear()
+    mm = mapdl.math
+    
+
+Parameters definition
+~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: ipython3
+
+    pi = np.arccos(-1)
+    rho = 1.2                        # density
+    c0 = 340                         # Speed of Sound
+    frq = 1000                       # excitation freq   Hz
+    visco = 0.9                      # viscosity
+    
+    TP = 1/frq
+    WL = c0 * TP
+    no_wl = 3                        # no of wavelengths in space
+    
+    cyl_L = no_wl * WL               # length of duct
+    cyl_r = 0.025 * cyl_L            # cross section of duct
+    
+    nelem_wl = 10                    # no of elements per wavelength
+    tol_elem = nelem_wl * no_wl
+
+Element and Material definition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: ipython3
+
+    mapdl.prep7()
+    
+    mapdl.et(1,'FLUID220', kop2=1)   # Uncoupled acoustic element without FSIs
+    mapdl.mp("DENS", 1, rho)
+    mapdl.mp("SONC", 1, c0)
+    mapdl.mp("VISC", 1, visco)
+
+.. parsed-literal::
+
+    MATERIAL          1     VISC =  0.9000000
+
+Geometry definition
+~~~~~~~~~~~~~~~~~~~
+
+.. code:: ipython3
+
+    mapdl.allsel()
+    mapdl.csys(0)
+    mapdl.wpcsys(-1)
+    mapdl.wprota(thzx=90)
+    mapdl.asel('NONE')
+    mapdl.vsel('NONE')
+    mapdl.cyl4(0, 0, cyl_r)
+    mapdl.wpcsys(-1)
+    mapdl.vext("ALL", dx=cyl_L)
+    mapdl.vsbw("ALL", keep='DELETE')
+    mapdl.wprota(thzx=90)
+    mapdl.vsbw("ALL", keep='DELETE')
+    mapdl.wpcsys(-1)
+    mapdl.cm('cm1', 'volu')
```

Comment:
Removed some unnecessary commands.

The two lines marked as to check if required is retained because we are cutting the cylinder into 4 segments for more uniform mesh. (Added description of the same ).